tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -12240,6 +12236,56 @@ static bool CheckVKBufferPointerCast(Sema &S, FunctionDecl *FD, CallExpr *CE,
 }
 #endif
 
+// Query function that returns the intersection-attribute parameter index
+// Returns -1 if the function doesn't have an intersection-attribute parameter
+static int GetIntersectionAttributeParamIdx(const FunctionDecl *FD) {
+  if (!FD)
+    return -1;
+
+  // Check if it's a HitObject member function
+  if (const CXXMethodDecl *Method = dyn_cast<CXXMethodDecl>(FD)) {
+    if (Method->getParent() && Method->getParent()->getName() == "HitObject") {
+      if (FD->getName() == "GetAttributes") {
```

Comment:
I believe this should be detectable through the same `HLSLIntrinsicAttr::getOpCode()` mechanism as `MOP_DxHitObject_GetAttributes`.  There should be no need to check the object and method names.